pkg/drain/nodegroup.go

```diff
@@ -144,18 +145,31 @@ func NodeGroup(clientSet kubernetes.Interface, ng *api.NodeGroup, waitTimeout ti
 
 			for _, node := range nodes.Items {
 				if newPendingNodes.Has(node.Name) {
+					retry := 1
 					pending, err := evictPods(drainer, &node)
-					if err != nil {
-						return err
-					}
-					logger.Debug("%d pods to be evicted from %s", pending, node.Name)
-					if pending == 0 {
-						drainedNodes.Insert(node.Name)
+					for pending > 0 {
+						logger.Debug("%d pods to be evicted from %s", pending, node.Name)
+						time.Sleep(5 * time.Second)
+						pending, err = evictPods(drainer, &node)
+						if err != nil && retry < podEvictionMaxRetries {
+							logger.Warning("pod eviction error: \"%s\", on node: %s (retry in 5 sec, %d/%d)", err, node.Name, retry, podEvictionMaxRetries)
```

Comment:
I think it help to use the same format we have for AWS API call retries:
```
logger.Warning("retryable error (%s) from %s - will retry after delay of %v", errorDescription, methodDescription, duration)
```

So I'd change it to this:
```suggestion
							logger.Warning("pod eviction error (%q) – will retry after delay of %v)", err, retryDelay)
```